3party/opening_hours/opening_hours.cpp

```diff
@@ -479,6 +489,22 @@ bool Timespan::HasPeriod() const
   return !m_period.IsEmpty();
 }
 
+bool Timespan::HasExtendedHours() const
+{
+  if (HasStart() && HasEnd() &&
```

Comment:
nit: How about to reduce indent in this method by negation of the condition? Like:
```c++
if (!HasStart() || !HasEnd() || ...)
{
  return false;
}
...
```
?